src/signal_scope/signaldata.cpp

```diff
@@ -185,6 +185,13 @@ QRectF SignalData::computeBounds()
     if (y > maxY) maxY = y;
   }
 
+  double marginMax = 1.1;
+  double marginMin = 1.1;
+  if (maxY < 0.0) marginMax = 0.9;
+  if (minY > 0.0) marginMin = 0.9;
+  maxY *= marginMax;
+  minY *= marginMin;
```

Comment:
Ok, that makes sense.  I think adding a margin to the plot Y limits is a reasonable feature to add, but it could be handled by the plot widget instead of the individual SignalData objects.